docker/inference/dockerfile.ctr

```diff
@@ -51,6 +53,19 @@ RUN pip install tritonclient[all] grpcio-channelz
 RUN pip install dask==2021.11.2 distributed==2021.11.2 dask[dataframe]==2021.11.2 dask-cuda
 RUN pip install git+https://github.com/rapidsai/asvdb.git@main
 
+
+# Triton Server
+FROM ${FULL_IMAGE} as full
+WORKDIR /opt/tritonserver
+COPY --chown=1000:1000 --from=full /opt/tritonserver/LICENSE .
+COPY --chown=1000:1000 --from=full /opt/tritonserver/TRITON_VERSION .
+COPY --chown=1000:1000 --from=full /opt/tritonserver/NVIDIA_Deep_Learning_Container_License.pdf .
+COPY --chown=1000:1000 --from=full /opt/tritonserver/bin bin/
+COPY --chown=1000:1000 --from=full /opt/tritonserver/lib lib/
+COPY --chown=1000:1000 --from=full /opt/tritonserver/include include/
+COPY --chown=1000:1000 --from=full /opt/tritonserver/repoagents/ repoagents/ 
+COPY --chown=1000:1000 --from=full /usr/bin/serve /usr/bin/.
```

Comment:
I think we will need to add these lines also in `dockerfile.torch`.

Reply:
@IamGianluca Triton server is already installed in pytorch and tensorflow inference containers 
```
albertoa@pursuit-dgxstation:~/Projects/Merlin/docker/inference$ docker run --pull always --gpus=all -it --ipc=host --cap-add SYS_NICE nvcr.io/nvstaging/merlin/merlin-pytorch-inference:22.03 /bin/bash
22.03: Pulling from nvstaging/merlin/merlin-pytorch-inference
Digest: sha256:8be045dfbb42ea128aca833b78f4847bcc69557ba97f5e65a3703f50606fc646
Status: Image is up to date for nvcr.io/nvstaging/merlin/merlin-pytorch-inference:22.03

=============================
== Triton Inference Server ==
=============================

NVIDIA Release 22.02 (build 32400308)

Copyright (c) 2018-2021, NVIDIA CORPORATION & AFFILIATES.  All rights reserved.

Various files include modifications (c) NVIDIA CORPORATION.  All rights reserved.

This container image and its contents are governed by the NVIDIA Deep Learning Container License.
By pulling and using the container, you accept the terms and conditions of this license:
https://developer.nvidia.com/ngc/nvidia-deep-learning-container-license

NOTE: Legacy NVIDIA Driver detected.  Compatibility mode ENABLED.

root@3f045a14f399:/opt/tritonserver# ls -la
total 12396
drwxr-xr-x  1 root          root             4096 Mar  4 10:17 .
drwxr-xr-x  1 root          root             4096 Feb 18 05:47 ..
-rw-rw-r--  1 triton-server triton-server    1485 Feb 18 01:23 LICENSE
-rw-rw-r--  1 triton-server triton-server 3012640 Feb 18 01:23 NVIDIA_Deep_Learning_Container_License.pdf
-rw-rw-r--  1 triton-server triton-server       7 Feb 18 01:23 TRITON_VERSION
drwxr-xr-x  1 triton-server triton-server    4096 Mar  4 10:17 backends
drwxr-xr-x  2 triton-server triton-server    4096 Feb 18 05:47 bin
drwxrwxr-x 15 root          root             4096 Mar  4 08:20 cmake-3.21.1
-rw-r--r--  1 root          root          9629567 Jul 27  2021 cmake-3.21.1.tar.gz
drwxr-xr-x  3 triton-server triton-server    4096 Feb 18 05:47 include
drwxr-xr-x  2 triton-server triton-server    4096 Feb 18 05:47 lib
-rwxrwxr-x  1 triton-server triton-server    4266 Feb 18 05:41 nvidia_entrypoint.sh
drwxr-xr-x  1 triton-server triton-server    4096 Feb 18 05:48 repoagents
root@3f045a14f399:/opt/tritonserver# ls -la bin/
total 10848
drwxr-xr-x 2 triton-server triton-server     4096 Feb 18 05:47 .
drwxr-xr-x 1 root          root              4096 Mar  4 10:17 ..
-rwxr-xr-x 1 triton-server triton-server 11092616 Feb 18 01:38 tritonserver
root@3f045a14f399:/opt/tritonserver#
```

This problems only happens in ctr (hugectr) since the base image is very minimal. I was working in getting smaller inference container since size was a problem for cloud providers, and I cut too much.

Reply:
Got it! Thank you for the explanation @albert17 :+1: